src/main/java/com/xmartlabs/jenkins/buildslacknotifier/BuildSlackNotifier.java

```diff
@@ -0,0 +1,106 @@
+package com.xmartlabs.jenkins.buildslacknotifier;
+
+import hudson.EnvVars;
+import hudson.Extension;
+import hudson.Launcher;
+import hudson.model.AbstractProject;
+import hudson.model.AbstractBuild;
+import hudson.model.BuildListener;
+import hudson.tasks.Recorder;
+import hudson.tasks.Publisher;
+import hudson.tasks.BuildStepDescriptor;
+import hudson.tasks.BuildStepMonitor;
+import hudson.util.FormValidation;
+
+import javax.servlet.ServletException;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import net.sf.json.JSONObject;
+
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.StaplerRequest;
+
+import com.xmartlabs.jenkins.buildslacknotifier.helper.GsonHelper;
+import com.xmartlabs.jenkins.buildslacknotifier.request.Attachment;
+import com.xmartlabs.jenkins.buildslacknotifier.request.Request;
+
+import org.kohsuke.stapler.QueryParameter;
+
+public class BuildSlackNotifier extends Recorder {
+  private List<SlackURL> urls = new ArrayList<>();
+
+  @DataBoundConstructor
+  public BuildSlackNotifier(ArrayList<SlackURL> urls) {
```

Comment:
Can't. For some reason it breaks the plugin (guessing something that Jenkins doesn't like)